### PR TITLE
Set Commit SHA as ENV variable

### PR DESCRIPTION
**Pull request by vigneshmsft**
Register OKComputer:AppVersionCheck, which checks for presense of ENV['SHA']
[`/check/version`](https://dev.apply-for-teacher-training.service.gov.uk/check/version) endpoint to respond with current version.
Set version info in Raven configuration for identifying version in Sentry.

## Context

Make it easy to identify offending versions in [Sentry](https://docs.sentry.io/platforms/ruby/configuration/releases/).
And the current version in each environment.

## Changes proposed in this pull request

`SHA` env variable set to the git commit (same as the docker image tag).

Adds optional [`OkComputer:AppVersionCheck`](https://github.com/sportngin/okcomputer/blob/master/lib/ok_computer/built_in_checks/app_version_check.rb#L4)

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

File: config/initializers/raven.rb [hunk 1]
```diff
@@ -1,6 +1,7 @@
 Raven.configure do |config|
   config.silence_ready = true
   config.current_environment = HostingEnvironment.environment_name
+  config.release = "#{HostingEnvironment.environment_name}@#{ENV['SHA']}"
```
Comment (tijmenb) [inline config/initializers/raven.rb:4]: I'm not sure if we should prefix this. In Sentry you can say "Resolve in next release", which means the error will only re-alert if it happens in a next version. If the version is hosting environment-specific, it won't work across environments.

**Review by davidgisbey — APPROVED** [review]

File: docker-compose.azure.yml [hunk 2]
```diff
@@ -6,6 +6,7 @@ services:
       args:
         bundleWithout: 'development'
         BASE_RUBY_IMAGE_WITH_GEMS_AND_NODE_MODULES: ${gemsNodeModulesImageName:-apply-for-teacher-training-gems-node-modules}
+        SHA: ${dockerHubImageTag}
```
Comment (tijmenb) [inline docker-compose.azure.yml:9]: Maybe something like `DEPLOYED_VERSION` instead of `SHA` would be clearer?

**Review by tijmenb — COMMENTED** [review]
Cool! Will help a lot with error tracking. One comment about the release we send to Sentry and one about the naming.

File: docker-compose.azure.yml (hunk 2 above)
Comment (vigneshmsft) [reply docker-compose.azure.yml:9]: began with `VERSION`, I'll change this to `VERSION` but realised OKComputer used `SHA` and changed all occurances,
will change this entry to `VERSION`

**Pull request merged**